visual indicator for new messages/chats

StevieSong opened the pull request:
per #135, new messages and chats needed a visual indicator on the admin side.

per the wiki, a 'dark-green' bar is shown on the user button in the left tab if there is an update regarding the user:
- a new user connected
- an existing user sent a new message
- an existing user disconnected

to get rid of the alert indicator, the admin can:
- accept an incoming thread
- send a message back
- delete the thread

chat messageStream object changed
- added 'alert' attribute that is a boolean to indicate if the alert indicator should be displayed in the user tab. change documented in wiki

other changes in this commit include renaming text input area messageBox -> inputBox to avoid confusion with message display area messagesBox

ajohns619 commented on public/javascript/admin.js:
```diff
@@ -127,14 +127,20 @@ function updateUserOverview() {
-                      + "<div class='messagePreview'>" + messagePreview + "</div></div>"
-                      + "<div class='buttonTypingDiv' " + userTypingHidden + ">"
-                      + "<img class='buttonTypingIcon' src='img/typing_icon.png'></div></button>";
+                        + iconTag
```
It looks like there's some mixing of tabs and spaces here, can you change them to all be spaces?

StevieSong replied:
was on purpose to make the nesting structure clearer but I can change it back to all left aligned '+' if you want

StevieSong replied:
(theyre all spaces)

ajohns619 replied:
Oh ok, that makes sense

ajohns619 commented:
Also, you might have to re-merge it into master, since Derek just pushed changes to add the "user typing" icon thing. We'll have to make sure that this icon doesn't overlap the green bar.

StevieSong commented:
> Also, you might have to resolve some new conflicts with master, since Derek just pushed changes to add the "user typing" icon thing. We'll have to make sure that this icon doesn't overlap the green bar.

done, changed the method by which we show/hide the typing icon and fixed the sizes so elements don't shift around
